src/lib/java/de/teamlapen/lib/lib/util/NotDrainableTank.java

```diff
@@ -0,0 +1,37 @@
+package de.teamlapen.lib.lib.util;
+
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.templates.FluidTank;
+
+import javax.annotation.Nonnull;
+import java.util.function.Predicate;
+
+public abstract class NotDrainableTank extends FluidTank {
```

Comment:
Name is quite misleading.
Could also be combined with the FluidTankWithListener